test/cpp/jit/test_gpu.cpp

```diff
@@ -18148,6 +18148,135 @@ TEST(NVFuserTest, FusionRfactorContigIDs_CUDA) {
+                        .linearize({32,1})])                 
+    =*reinterpret_cast<uint4*>(Bg_buffer);
+    // Read from Smem:
+    __half A[16];
+    __half B[16];
+    // Two 128b reads per operand, this loop to be 
+    //  generated by scheduling
+    for(int warptile_i = 0;warptile_i<2;warptile_i++){
+        *reinterpret_cast<uint4*>(&A[8*warptile_i]) =
+        *reinterpret_cast<uint4*>(
+            &As[
+                mem_swizzle::swizzle_sequence<
+                    warp_mma::mmaM32N32K8::SmemReadASwizzle>(
+                        // Need to skip iterdomains within
+                        //  instruction tile in this indexing
+                        //  since they will be determined by lane_id
+                        {16*warptile_i+threadIdx.x, 0}, lane_id
+                    ).linearize({8,1})
+            ]);
+        
+        // This will be a separate loop in the generated code
+        //   but both will be unrolled eventually.
+        *reinterpret_cast<uint4*>(&B[8*warptile_i]) =
+        *reinterpret_cast<uint4*>(
+            &Bs[
+                mem_swizzle::swizzle_sequence<
+                    warp_mma::mmaM32N32K8::SmemReadBSwizzle>(
+                        // Need to skip iterdomains within
+                        //  instruction tile in this indexing
+                        //  since they will be determined by lane_id
+                        {4*warptile_i, 0}, lane_id
+                    ).linearize({32,1})
+            ]);
+    }
+    // finally calling the mma:
+    warp_mma::mmaM32N32K8::run(&C[0], &A[0], &B[0]);
+    // Write back result:
+    
+    // write back option 1: use un-swizzle helper
+    for(int write_c_i=0;write_c_i<32;write_c_i++){
+      Cg1[
+        warp_mma::mmaM32N32K8::UnswizzleFloatC::run(
+          {0, write_c_i},
+          lane_id
+        ).linearize({(int)Cg1.stride[0],(int)Cg1.stride[1]})
+      ] = C[write_c_i];
+    }
+    // write back option 2: using the writeC helper
+    warp_mma::mmaM32N32K8::writeC(&Cg0[0], &C[0], lane_id, Cg0.stride[0]);
+  }
+    )";
+  fe.compileRtc(kernel, "CudaCodeGen::mma32x32x8_mini_gemm_float");
+  LaunchParams lp(
+      1, // gdimx
+      1, // gdimy
+      1, // gdimz
+      32, // bdimx
+      1, // bdimy
+      1 // bdimz
+  );
+  lp.setSmem(0);
+  const auto options =
+      at::TensorOptions().dtype(at::kHalf).device(at::kCUDA, 0);
+  auto inA = at::randn({32, 8}, options);
+  auto inB = at::randn({8, 32}, options);
+  auto outC0 = at::empty({32, 32}, options.dtype(at::kFloat));
+  auto outC1 = at::empty({32, 32}, options.dtype(at::kFloat));
+
+  fe.runRtc(lp, {inA, inB, outC0, outC1});
+
+  auto refC = inA.matmul(inB).to(at::kFloat);
+
+  // Manually set tolerance for now, using the half
+  //  reduction tolerance. half reduction to float
+  //  doesn't reach full float tolerance but is better
+  //  than accumulation into half .
+  TORCH_CHECK(refC.allclose(outC0, 0.001, 0.0001));
```

Comment:
I tried the reference version in double precision, and the check passed with the default tolerance.

```
 auto refC = inA.to(at::kDouble).matmul(inB.to(at::kDouble)).to(at::kFloat);
```

Reply:
Thanks for pointing this out. Sorry I didn't put this on the comment. This mini gemm case does pass with float/double version of cublas as reference. I could use the test_validate here if we want. But it failed again with larger k (on the gemm integration PR). It didn't seem to be able to consistently meet our precision targets, so I just used manually set ones for everything.

Reply:
I will change the reference to use the double version though. Wouldn't want cublas to accumulate in half.

Reply:
Generally our tolerances are based on comparisons with a double precision counter part. Comparing float-float requires much larger tolerances because of the high variance float results can have, so leaving the comparison in double allows us to tighten our validation tolerances which is quite helpful.